Assets/Scripts/Utility/UIManager.cs

```diff
@@ -0,0 +1,109 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+
+public class UIManager : MonoBehaviour
+{
+    // Var Player and Abilities
+    private GameObject player;
+    private GameObject abilities;
+    // Var HP
+    private GameObject playerHPBar;
+    private Image hpFilling;
+    private TMP_Text hpText;
+    // Var Mana
+    private GameObject playerManaBar;
+    private Image manaFilling;
+    private TMP_Text manaText;
+    // Var Teleport
+    private GameObject teleportObject;
+    private Image teleportShadow;
+    private float teleportTimer;
+    private float teleportCooldown;
+    // Var PierceAttack
+    private GameObject pierceAttackObject;
+    private Image pierceAttackShadow;
+    private float pierceAttackTimer;
+    private float pierceAttackCooldown;
+    // Var BasicAttack
+    private GameObject basicAttackObject;
+    private Image basicAttackShadow;
+    private float basicAttackTimer;
+    private float basicAttackCooldown;
+
+
+    void Start()
+    {
+        //Access to HP Bar
+        playerHPBar = transform.Find("HPBar").gameObject;
+        hpFilling = playerHPBar.transform.GetChild(1).GetComponent<Image>();
+        hpText = playerHPBar.transform.GetChild(2).GetComponent<TMP_Text>();
+        // Access to MP Bar
+        playerManaBar = transform.Find("MPBar").gameObject;
+        manaFilling = playerManaBar.transform.GetChild(1).GetComponent<Image>();
+        manaText = playerManaBar.transform.GetChild(2).GetComponent<TMP_Text>();
+        // Access To Player and Abilities
+        player = GameObject.FindGameObjectWithTag("Player");
```

Comment:
the reason you want the player is because what actually want is the ManaPoints and the PlayerHealthPoints. make them variables up top and SerializeField them so you can assign them in the inspector.